app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoDumper.java

```diff
@@ -165,10 +165,8 @@ private InputStream toStream() throws IOException {
             serializer.setOutput(outputStream, XML_ENCODING);
             serializer.startDocument(XML_ENCODING, true);
             serializer.setFeature("http://xmlpull.org/v1/doc/features.html#indent-output", true);
-            final UiElement<?, ?> xpathRoot = root == null
-                    ? rebuildForNewRoot(currentActiveWindowRoot(), NotificationListener.getInstance().getToastMessage())
-                    : rebuildForNewRoot(root, null);
-            serializeUiElement(xpathRoot, 0);
+            final UiElement<?, ?> xpathRoot = root == null ? getCachedRoot() : getCachedElement(root);
```

Comment:
Looks like we are ignoring ToastMessage?

Reply:
this has been changed